### PR TITLE
Updated to swift 4. Removed secp context leaking over 1mb.

### PR DESCRIPTION
**Pull request by elland**
You read that right, over 1mb. 😭

### CODE REVIEW

File: EtherealCereal/EtherWrapper.m [hunk 1]
```diff
@@ -60,6 +62,8 @@ - (NSString * _Nonnull)signMessage:(NSData * _Nonnull)message withKey:(NSData *
     NSData *signatureData = [NSData dataWithBytes:&signature_serialised_with_recid length:65];
     NSString *signatureString = [signatureData hexadecimalString];
 
+    secp256k1_context_destroy(ctx);
```
Comment (elland) [inline EtherealCereal/EtherWrapper.m:65]: Never deallocated, created on every `signMessage` call. Ouch!

File: EtherealCereal/EtherealCereal.swift [hunk 2]
```diff
@@ -64,8 +64,9 @@ public class EtherealCereal: NSObject {
 
         // This creates the private key inside a block, result is of internal type ResultType.
         // We just need to check if it's 0 to ensure that there were no errors.
+        let count = privateKey.count
```
Comment (designatednerd) [inline EtherealCereal/EtherealCereal.swift:67]: why pull this out as a separate var?

File: EtherealCereal/EtherealCereal.swift [hunk 3]
```diff
@@ -130,28 +131,31 @@ struct EtherealCerealRegex {
...
 public extension String {
     public var hexadecimalData: Data? {
-        guard let match = EtherealCerealRegex.hexadecimalDataRegex.matches(in: self, range: NSMakeRange(0, self.count)).first
-            else { return nil }
-
-        let hexadecimalString = (self as NSString).substring(with: match.range(at: 1))
-        let utf16View: UTF16View
-        if hexadecimalString.count % 2 == 1 {
-            utf16View = "0\(hexadecimalString)".utf16
+        let str: String
```
Comment (designatednerd) [inline EtherealCereal/EtherealCereal.swift]: can you please use a full word for variable names per code standards?

File: EtherealCereal/EtherWrapper.m (hunk 1 above)
Comment (yuliaveres) [inline EtherealCereal/EtherWrapper.m:65]: 👍

File: EtherealCereal/EtherWrapper.m [hunk 4]
```diff
@@ -46,7 +48,7 @@ - (NSString * _Nonnull)signMessage:(NSData * _Nonnull)message withKey:(NSData *
 
     const unsigned char *private_key = privateKey.bytes;
     int recid = 0;
-    const secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
+    secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
```
Comment (yuliaveres) [inline EtherealCereal/EtherWrapper.m:51]: do you remember why was it const @Elland  ?

Comment (elland) [reply EtherealCereal/EtherWrapper.m:51]: Yes, I just wanted to be sure it wasn't changed, but then it is changed when we pass it to the destroy method, so `const` was being ignored by the compiler.

File: EtherealCereal.xcodeproj/xcshareddata/xcschemes/Tests.xcscheme [hunk 5]
```diff
@@ -1,58 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yuliaveres) [inline EtherealCereal.xcodeproj/xcshareddata/xcschemes/Tests.xcscheme]: deleted tests scheme ?

**Comment by designatednerd** [comment]
@Elland does this not have any tests, or is Circle just refusing to run them?

File: EtherealCereal/EtherealCereal.swift (hunk 2 above)
Comment (elland) [reply EtherealCereal/EtherealCereal.swift:67]: Compilar complains about referencing `privateKey` inside that block. 😀

File: EtherealCereal.xcodeproj/xcshareddata/xcschemes/Tests.xcscheme (hunk 5 above)
Comment (elland) [reply EtherealCereal.xcodeproj/xcshareddata/xcschemes/Tests.xcscheme]: Built inside the framework scheme instead. 🚀

File: EtherealCereal/EtherealCereal.swift (hunk 3 above)
Comment (elland) [reply EtherealCereal/EtherealCereal.swift]: Quite sure `string` is not any clearer than `str` here. Considering 90% of the codebase is in C, not sure this is much of a problem. 🤔

File: EtherealCereal/EtherealCereal.swift (hunk 2 above)
Comment (designatednerd) [reply EtherealCereal/EtherealCereal.swift:67]: ah yeah that makes sense

File: EtherealCereal/EtherealCereal.swift (hunk 3 above)
Comment (designatednerd) [reply EtherealCereal/EtherealCereal.swift]: This isn't C though :trollface: 

Also, you could use something like `hexString`

Comment (elland) [reply EtherealCereal/EtherealCereal.swift]: This isn't a hex string tho…

**Review by yuliaveres — APPROVED** [review]

**Comment by designatednerd** [comment]
I'm not actually seeing hte changes from your last commit in here @Elland - did you just put it back to something it was?

**Comment by elland** [comment]
Not sure what you're referring to. 🤔 Github still shows me the changes (destroying the `*ctx`, etc).

**Comment by designatednerd** [comment]
@Elland these changes: https://github.com/bakkenbaeck/EtherealCereal/pull/10/commits/42aa6d49d494c1c4e50911cbd82554ff0bb4f32b

**Comment by elland** [comment]
Yeah, turns out my master was outdated, and those swift4 updates were not needed.

**Review by designatednerd — APPROVED** [review]
:shipit:

**Pull request merged**